### ISSUE TITLE
What does e=1e-06 in peelingInfo.segregationTensorXY = ProbMath.generateSegregationXYChrom(e=1e-06) mean?

### ISSUE THREAD

**Issue by gregorgorjanc** [AlphaGenes/AlphaPeel#123]
Looking at recent PR https://github.com/AlphaGenes/AlphaPeel/pull/122/files I noticed this line of code

```
peelingInfo.segregationTensorXY = ProbMath.generateSegregationXYChrom(e=1e-06)
```

@XingerTang and @AprilYUZhang please have a look at what is the meaning of `e=1e-06` as in, is this another "error/noise" rate that we would like to control via an argument or is this just a numerical convenience to start a tensor with probabilities that are not zero?

**Comment by gregorgorjanc** [comment]
That’s mutation rate (too high as most mutation rates are ~10^(-8) or so). It’s used to create a transmission function between parents and a progeny.

**Comment by AprilYUZhang** [comment]
I am not sure if it is for numerical convenience, but it is hard to explain by the mutation rate. The first reason is if we consider the error when we deal with the transmission process. There are four possible sources, the father's haplotype (or allele), the mother's haplotype (or allele), the child 's haplotype, and segregation (this might be caused by pedigree error). At least three sources are related to mutation rate; it should not be added like this `segregationTensor = segregationTensor*(1-e) + e/4 ` in code; instead, it should be more complex.

**Comment by gregorgorjanc** [comment]
@AprilYUZhang it’s actually quite simple and doable.

Traditional transmission function works by having father's genotype as input, mother's genotype as input, and then progeny child as output.

For a given true genotype of father (say aa) and true genotype of mother (say) we build probability distribution of the progeny. When there is no mutation, the conditional probability distribution for the child given parental genotype is in this case p(aa, aA, Aa, AA | father = aa, mother = aa) = [1, 0, 0, 0].

Now let's expand this a bit and consider mutation rate of e. First, let's work out possible cases:
* father's alleles are a and a and he can pass to progeny
  * first allele (a) as a with probability 1-e and as A with probability (e)
  * second allele (a) as a with probability 1-e and as A with probability (e)
* mother' alleles are a and a and she can pass to progeny
  * first allele (a) as a with probability 1-e and as A with probability (e)
  * second allele (a) as a with probability 1-e and as A with probability (e)
 
We now need to pair these 4x4 probabilities to obtain 16 possible combinations with 4 states (aa, aA, Aa, AA) which leads to p(aa, aA, Aa, AA | father = aa, mother = aa) = [(1-e)^2, (1-e)e, e(1-e), e^2]. So not hard, just tedious AND we need to do it from all possible pairs genotypes in parents!

The transmission function in AlphaPeel uses another dimension, so instead of the above 3-dimensional distribution, we have to work with 4-dimensional distribution, but again the key is that inputs are father's genotype, mother's genotype, AND segregation status, and output variable is progeny genotype. To manage this, we simply have to look at a p(aa, aA, Aa, AA | father = aa, mother = aa, segregation = pp) = [1, 0, 0, 0], which is not very informative, but check this out p(aa, aA, Aa, AA | father = aA, mother = aA, segregation = pp) = [1, 0, 0, 0] - do you see how setting segregation to pp it means that we know which of the two parent's alleles was inherited by the progeny (paternal from father and paternal from mother), while p(aa, aA, Aa, AA | father = aA, mother = aA, segregation = pm) = [0, 1, 0, 0].

Does this make sense?

We just need to add e into the above segregation-informed transmission function.

**Comment by AprilYUZhang** [comment]
Yes, I understand your meaning. I mean, Andrew adding this 'e' may not be caused by mutation rate. If we add this, it is feasible.
